tests_e2e/tests/agent_publish/agent_publish.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import uuid
+from typing import Any, Dict, List
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.identifiers import VmExtensionIds, VmExtensionIdentifier
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+
+
+class AgentPublishTest(AgentTest):
+    """
+    This script verifies if the agent update performed in the vm.
+    """
+
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client: SshClient = self._context.create_ssh_client()
+
+    def run(self):
+        """
+        we run the scenario in the following steps:
+            1. Print the current agent version before the update
+            2. Prepare the agent for the update
+            3. Check for agent update from the log
+            4. Print the agent version after the update
+            5. Ensure CSE is working
+        """
+        self._get_agent_info()
+        self._prepare_agent()
+        self._check_update()
+        self._get_agent_info()
+        self._check_cse()
+
+    def _get_agent_info(self) -> None:
+        stdout: str = self._ssh_client.run_command("waagent-version", use_sudo=True)
+        log.info('Agent info \n%s', stdout)
+
+    def _prepare_agent(self) -> None:
+        output = self._ssh_client.run_command("modify-agent-publish-config", use_sudo=True)
+        log.info('Modifying agent update related config flags \n%s', output)
```

Comment:
We should log this info before invoking the remote command. If the command fails, the call will raise an exception and the "Modifying..." message will be lost. Of course, the output needs to be logger after the command completes.

Reply:
same for other remote commands